Userland/DynamicLoader/CMakeLists.txt

```diff
@@ -3,6 +3,8 @@ set(LOADER_SOURCES
     misc.cpp
 )
 
+set(LIBCORE_SOURCES "../Libraries/LibCore/ArgsParser.cpp")
```

Comment:
No, no, and no. If we *really* want to have LibCore in dynamic loader, let's do the same thing I did in #24105 and have some kind of DynamicLoader_LibCoreMinimal.

Reply:
Please see this comment:
https://github.com/SerenityOS/serenity/pull/21101#pullrequestreview-1629985941
I was explicitly asked to do this, although I was very interested in making arg parsing a library on its own.
So until there's another idea to settle this, I rather go with what @BertalanD says which makes sense to me.

Reply:
I've mostly meant doing something like
```cmake
add_library(DynamicLoader_LibCoreArgsParser ArgsParser.cpp)
target_link_libraries(DynamicLoader_LibCoreArgsParser PUBLIC DynamicLoader_CompileOptions)
```
in LibCore/CMakeLists.txt and having
```cmake
target_link_libraries(DynamicLoader PRIVATE DynamicLoader_LibCoreArgsParser)
```
in DynamicLoader/CMakeLists.txt. This way we don't spread LibCore all over Serenity.

P. S.: "No, no, and no." might indeed be a bit too excessive but I don't see how the rest of the review has a wrong tone.

Reply:
Done. I also converted the `Version.cpp` code to use LibC `uname()` function, as we can't directly call any `Core::System` methods with this approach (which is OK).

P. S.: "No, no, and no." is indeed excessive (imo). the rest of the review has a no wrong tone, I just didn't like the opening of your comment. You can dismiss my advice, but I suggest to write to the point - I never ever bluntly dismiss well-written comments and in particular a good technical advice as I really like to give the best-shaped piece of code I can write. Giving an example as you did in the second comment on the topic was well-written and I immediately understood what you mean. That's just my two cents on the matter.